--- profiler_api.inc 2011-05-06 12:35:07.000000000 -0400
+++ profiler_api.inc 2011-05-06 12:35:07.000000000 -0400
@@ -443,6 +443,26 @@ 
       }
     }

+    // Detect any taxonomy and handle those
+    $taxonomy = array();
+    foreach ($properties['taxonomy'] as $vocab => $terms) {
+      if (is_string($vocab)) {
+        $vocab = trim($vocab);
+        $vid = db_result(db_query("SELECT vid FROM {vocabulary} WHERE name='%s'", $vocab));
+        if ($vid != NULL && is_string($terms)) { 
+          $terms = explode(',', $terms); 
+          foreach($terms as $term) {
+            $term = trim($term);
+            $tid = db_result(db_query("SELECT tid FROM {term_data} WHERE vid=%d AND name='%s'", $vid, $term));
+            if($tid != NULL) {
+              $taxonomy[$vid][$tid] = $tid;
+            }
+          }
+        }
+      }
+    }
+    $node->taxonomy = $taxonomy;
+
     node_save($node);
   }
 }